### PR TITLE
[supervisor] Improve IDE ReadinessHTTPProbe

### PR DESCRIPTION
**Pull request by aledbf**
We can see in the log the time, like `"IDE readiness took 8.021 seconds",`

### CODE REVIEW

File: components/supervisor/pkg/supervisor/supervisor.go [hunk 1]
```diff
@@ -567,24 +567,28 @@ func runIDEReadinessProbe(cfg *Config) {
...
-			tick   = time.NewTicker(1 * time.Second)
+			client = http.Client{Timeout: 1 * time.Second}
+			tick   = time.NewTicker(500 * time.Millisecond)
 		)
 		defer tick.Stop()
+
+		t0 := time.Now()
+
 		for {
+			<-tick.C
```
Comment (csweichel) [inline components/supervisor/pkg/supervisor/supervisor.go:578]: placing the tick here means we might be 500 milliseconds slower than we need to be if the IDE is already available by the time we start probing.

**Review by csweichel — APPROVED** [review]

File: components/supervisor/pkg/supervisor/supervisor.go (hunk 1 above)
Comment (aledbf) [reply components/supervisor/pkg/supervisor/supervisor.go:578]: yes, I am aware :)

**Review by csweichel — APPROVED** [review]

**Pull request merged**